data-products/src/main/scala/org/sunbird/obsrv/dataproducts/job/MasterDataProcessorIndexer.scala

```diff
@@ -0,0 +1,156 @@
+    val provider = providerFormat(config.getString("cloudStorage.provider"))
+    val inputSourceSpec = if(provider.druidProvider == "local")
+      s"""{"spec":{"ioConfig":{"type":"index_parallel","inputSource":{"type":"${provider.druidProvider}","baseDir":"${config.getString("cloudStorage.container")}","filter":"**.json.gz"}}}}"""
+    else s"""{"spec":{"ioConfig":{"type":"index_parallel","inputSource":{"type":"${provider.druidProvider}","objectGlob":"**.json.gz","prefixes":["$filePath"]}}}}"""
+    val deltaJson = parse(deltaIngestionSpec)
+    val inputSourceJson = parse(inputSourceSpec)
+    val ingestionSpec = parse(datasource.ingestionSpec)
+    val modIngestionSpec = ingestionSpec merge deltaJson merge inputSourceJson
+    compact(render(modIngestionSpec))
+  }
+
+  private def providerFormat(cloudProvider: String): BlobProvider = {
+    cloudProvider match {
+      case "local" => BlobProvider("file", "local", "file")
+      case "aws" => BlobProvider("s3a","s3", "s3")
+      case "azure" => BlobProvider("wasbs", "azure", "azure")
+      case "gcloud" => BlobProvider("gs", "google", "gs")
+      case "cephs3" => BlobProvider("s3a", "s3", "s3") // TODO: Have to check Druid compatibility
+      case "oci" => BlobProvider("s3a", "s3", "s3") // TODO: Have to check Druid compatibility
+      case _ => throw new Exception("Unsupported provider")
+    }
+  }
+
+  private def submitIngestionTask(ingestionSpec: String) = {
+    // TODO: Handle success and failure responses properly
+    val response = Unirest.post(config.getString("druid.indexer.url"))
+      .header("Content-Type", "application/json")
+      .body(ingestionSpec).asJson()
+    logger.info("Ingestion spec - " + response.getBody)
+    response.ifFailure(response => throw new Exception(s"Exception while submitting ingestion task - ${response.getBody}"))
+  }
+
+  private def deleteDataSource(datasourceRef: String): Unit = {
+    // TODO: Handle success and failure responses properly
+    val response = Unirest.delete(config.getString("druid.datasource.delete.url") + datasourceRef)
+      .header("Content-Type", "application/json")
+      .asJson()
+    response.ifFailure(response => throw new Exception("Exception while deleting datasource" + datasourceRef + " ,Response body - " + response.getBody))
+  }
+
+  def createDataFile(dataset: Dataset, outputFilePath: String) = {
+    val conf = new SparkConf()
+      .setAppName("MasterDataProcessorIndexer")
+      .set("spark.redis.host", dataset.datasetConfig.redisDBHost.get)
+      .set("spark.redis.port", String.valueOf(dataset.datasetConfig.redisDBPort.get))
+      .set("spark.redis.db", String.valueOf(dataset.datasetConfig.redisDB.get))
+    val readWriteConf = ReadWriteConfig(scanCount = 1000, maxPipelineSize = 1000)
```

Comment:
Can we read these scanCount and maxPipelineSize properties from the configurations ?

Reply:
yea sure.